batchproducer/batchproducer.go

```diff
@@ -187,15 +200,21 @@ func (b *batchProducer) Add(data []byte, partitionKey string) error {
 
 // from/for interface Producer
 func (b *batchProducer) Start() error {
-	if b.isRunning() {
-		return nil
+	b.runningMu.Lock()
+	defer b.runningMu.Unlock()
```

Comment:
> The change I proposed, to reduce contention, looks ugly.

Ah, got it.

> As for straightforwardness, I guess you're right. (Sadly,) Atomics aren't as widely known as locks.

:neutral_face: 

> Anyway, this is personal opinion. I always go for the least contention possible.

Well we all have our quirks ;)

> This code is good to merge.

The whole PR?